core-web/libs/template-builder/src/lib/components/template-builder/template-builder.component.ts

```diff
@@ -413,7 +162,46 @@ export class TemplateBuilderComponent implements OnInit, AfterViewInit {
         });
     }
 
-    identify(index: number, w: GridStackWidget) {
+    ngOnDestroy(): void {
+        this.grid.destroy(true);
+    }
+
+    identify(_: number, w: GridStackWidget) {
         return w.id;
     }
+
+    /**
+     * @description This is called when a widget is dropped in a subgrid
+     *
+     * @private
+     * @param {GridStackNode} oldNode This is not undefined when you dropped a widget that was on another subGrid
+     * @param {GridStackNode} newNode This is the newNode that was dropped
+     * @memberof TemplateBuilderComponent
+     */
+    private subGridOnDropped(oldNode: GridStackNode, newNode: GridStackNode) {
+        // If the oldNode exists, then the widget was dropped from another subgrid
+        if (oldNode && newNode) {
+            this.store.moveColumnInYAxis(
+                [oldNode, newNode].map((node, i) => ({
+                    parentId: node.grid?.parentGridItem?.id as string,
+                    w: node.w,
+                    h: node.h,
+                    x: node.x,
+                    y: node.y,
+                    id: i ? String(ids++) : node.id
+                }))
+            );
+        } else {
+            this.store.addColumn({
+                parentId: newNode.grid?.parentGridItem?.id as string,
+                w: newNode.w,
+                h: newNode.h,
+                x: newNode.x,
+                y: newNode.y,
+                id: String(ids++)
+            });
+
+            newNode.grid?.removeWidget(newNode.el as GridStackElement, true);
+        }
```

Comment:
Let's leave the store take care of this, we send the whole thing and the store (or some external util) take care of this. Let's keep the component as dumb as possible.